service/src/test/java/greencity/service/ubs/SuperAdminServiceImplTest.java

```diff
@@ -859,4 +865,382 @@ void changeTariffLocationsStatusParamUnresolvable() {
         assertThrows(BadRequestException.class,
             () -> superAdminService.changeTariffLocationsStatus(1L, dto, "unresolvable"));
     }
-}
+
+    @Test
+    void deactivateTariffByOneRegion() {
+        DetailsOfDeactivateTariffsDto details = ModelUtils.getDetailsOfDeactivateTariffsDtoWithRegion();
+
+        when(deactivateTariffsForChosenParamRepository.isRegionsExists(anyList())).thenReturn(true);
+        superAdminService.deactivateTariffForChosenParam(details);
+        verify(deactivateTariffsForChosenParamRepository).isRegionsExists(List.of(1L));
+        verify(deactivateTariffsForChosenParamRepository).deactivateTariffsByRegions(List.of(1L));
+        verifyNoMoreInteractions(deactivateTariffsForChosenParamRepository);
+    }
+
+    @Test
+    void deactivateTariffByTwoRegions() {
+        DetailsOfDeactivateTariffsDto details = ModelUtils.getDetailsOfDeactivateTariffsDtoWithRegion();
+        details.setRegionsId(Optional.of(List.of(1L, 2L)));
+
+        when(deactivateTariffsForChosenParamRepository.isRegionsExists(anyList())).thenReturn(true);
+        superAdminService.deactivateTariffForChosenParam(details);
+        verify(deactivateTariffsForChosenParamRepository).isRegionsExists(List.of(1L, 2L));
+        verify(deactivateTariffsForChosenParamRepository).deactivateTariffsByRegions(List.of(1L, 2L));
+        verifyNoMoreInteractions(deactivateTariffsForChosenParamRepository);
+    }
+
+    @Test
+    void deactivateTariffByNotExistingRegionThrows() {
+        DetailsOfDeactivateTariffsDto details = ModelUtils.getDetailsOfDeactivateTariffsDtoWithRegion();
+
+        when(deactivateTariffsForChosenParamRepository.isRegionsExists(anyList())).thenReturn(false);
+        assertThrows(NotFoundException.class,
+            () -> superAdminService.deactivateTariffForChosenParam(details));
+        verify(deactivateTariffsForChosenParamRepository).isRegionsExists(List.of(1L));
+        verifyNoMoreInteractions(deactivateTariffsForChosenParamRepository);
+    }
+
+    @Test
+    void deactivateTariffByOneRegionAndCities() {
+        DetailsOfDeactivateTariffsDto details = ModelUtils.getDetailsOfDeactivateTariffsDtoWithRegionAndCities();
+
+        when(regionRepository.existsRegionById(anyLong())).thenReturn(true);
+        when(deactivateTariffsForChosenParamRepository.isCitiesExistForRegion(anyList(), anyLong())).thenReturn(true);
+        superAdminService.deactivateTariffForChosenParam(details);
+        verify(regionRepository).existsRegionById(1L);
+        verify(deactivateTariffsForChosenParamRepository).isCitiesExistForRegion(List.of(1L, 11L), 1L);
+        verify(deactivateTariffsForChosenParamRepository).deactivateTariffsByRegionsAndCities(List.of(1L, 11L), 1L);
+        verifyNoMoreInteractions(regionRepository, deactivateTariffsForChosenParamRepository);
```

Comment:
please move this verification to the `After` method and add all mocks as input params

Reply:
done